diff --git a/src/exports/vite/dev/index.js b/src/exports/vite/dev/index.js
index e3e69432aa81932fff071f21dd44da6a45cdc76a..24fbcbdef60e6e9b1bc806496326ff21ee4f45aa 100644
--- a/src/exports/vite/dev/index.js
+++ b/src/exports/vite/dev/index.js
@@ -202,7 +202,7 @@ export async function dev(vite, vite_config, svelte_config) {
 								if (
 									(isCSSRequest(dep.file) ||
 										(query.has('svelte') && query.get('type') === 'style')) &&
-									!(query.has('raw') || query.has('url') || query.has('inline'))
+										!(query.has('raw') || query.has('url') || query.has('inline') || query.has('noinline'))
 								) {
 									try {
 										query.set('inline', '');
